Sources/FoundationInternationalization/ICU/ICU+StringConverter.swift

```diff
@@ -0,0 +1,204 @@
+            defer {
+                ucnv_resetFromUnicode(converter)
+            }
+
+            let utf16Rep = string.utf16
+            let uchars = UnsafeMutableBufferPointer<UChar>.allocate(capacity: utf16Rep.count)
+            _ = uchars.initialize(fromContentsOf: utf16Rep)
+            defer {
+                uchars.deallocate()
+            }
+
+            let srcLength = uchars.count
+            let capacity = srcLength * Int(ucnv_getMaxCharSize(converter)) + 1
+            let dest = UnsafeMutableRawPointer.allocate(
+                byteCount: capacity,
+                alignment: MemoryLayout<CChar>.alignment
+            )
+
+            var status: UErrorCode = U_ZERO_ERROR
+            if lossy {
+                var lossyChar: UChar = encoding == .ascii ? 0xFF : 0x3F
+                ucnv_setSubstString(
+                    converter,
+                    &lossyChar,
+                    1,
+                    &status
+                )
+                guard status.isSuccess else { return nil }
+
+                ucnv_setFromUCallBack(
+                    converter,
+                    UCNV_FROM_U_CALLBACK_SUBSTITUTE,
+                    nil, // newContext
+                    nil, // oldAction
+                    nil, // oldContext
+                    &status
+                )
+                guard status.isSuccess else { return nil }
+            } else {
+                ucnv_setFromUCallBack(
+                    converter,
+                    UCNV_FROM_U_CALLBACK_STOP,
+                    nil, // newContext
+                    nil, // oldAction
+                    nil, // oldContext
+                    &status
+                )
+                guard status.isSuccess else { return nil }
+            }
+
+            let actualLength = ucnv_fromUChars(
+                converter,
+                dest,
+                CInt(capacity),
+                uchars.baseAddress,
+                CInt(srcLength),
+                &status
+            )
+            guard status.isSuccess else { return nil }
+            return Data(
+                bytesNoCopy: dest,
+                count: Int(actualLength),
+                deallocator: .custom({ pointer, _ in pointer.deallocate() })
+            )
+        }
+    }
+}
+
+extension ICU.StringConverter {
+    private static let _converters: LockedState<[String.Encoding: ICU.StringConverter]> = .init(initialState: [:])
+
+    static func converter(for encoding: String.Encoding) -> ICU.StringConverter? {
+        return _converters.withLock {
+            if let converter = $0[encoding] {
+                return converter
+            }
+            if let converter = ICU.StringConverter(encoding: encoding) {
+                $0[encoding] = converter
+                return converter
+            }
+            return nil
+        }
+    }
+}
+
+
+@_dynamicReplacement(for: _icuMakeStringFromBytes(_:encoding:))
```

Comment:
This replaced function is only defined in `!FOUNDATION_FRAMEWORK` builds, could we wrap this in `#if !FOUNDATION_FRAMEWORK` as well so that it will build successfully in `FOUNDATION_FRAMEWORK` mode?

Reply:
Done!